GIP/GeoAlgorithms.cpp

```diff
@@ -256,11 +256,14 @@ namespace gip {
 
         // create output
         // convert extent to resolution units
-        int xsz = std::ceil(ext.width() / std::abs(xres));
-        int ysz = std::ceil(ext.height() / std::abs(yres));
+        int xsz = std::ceil((ext.width() + std::abs(xres)) / std::abs(xres));
+        int ysz = std::ceil((ext.height()+ std::abs(yres))/ std::abs(yres));
 
-        CImg<double> bbox(4,1,1,1, ext.x0(), ext.y0(), ext.width(), ext.height());
-        GeoImage imgout = GeoImage::create(filename, xsz, ysz, geoimgs[0].nbands(), 
+        double xshift = -0.5 * std::abs(xres);
+        double yshift = 0.5  * std::abs(yres);
+
+        CImg<double> bbox(4,1,1,1, ext.x0() + xshift, ext.y0() - yshift, xsz * std::abs(xres), ysz * std::abs(yres));
```

Comment:
Can you add a comment here explaining the idea behind this new bounding box, for future confused humans.